hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/Invocation.java

```diff
@@ -580,9 +579,7 @@ private void handleRetry(Object cause) {
 
         if (future.interrupted) {
             complete(INTERRUPTED);
-        } else {
-            context.invocationRegistry.deregister(this);
-
+        } else if (op.getCallId() == 0 || context.invocationRegistry.deregister(this)) {
```

Comment:
`op.getCallId() == 0` is for the case invocation failed even before registering, right?